util/stopwaiter/stopwaiter.go

```diff
@@ -73,9 +87,36 @@ func (s *StopWaiterSafe) StopOnly() {
 }
 
 // Stopping multiple times, even before start, will work
-func (s *StopWaiterSafe) StopAndWait() {
+func (s *StopWaiterSafe) StopAndWait() error {
+	return s.stopAndWaitImpl(stopDelayWarningTimeout)
+}
+
+func getAllStackTraces() string {
```

Comment:
Maybe better to use `debug.Stack()`

Reply:
I believe that one's only for the current thread